### DISCUSSION TITLE
Using default traitlet values from the Javascript side

### DISCUSSION THREAD

**Discussion by psychemedia** [manzt/anywidget#640] [General]
When setting initial values for traitlets / shared state across the python and js sides, the initial/default value seems to inherit from the pyhton side.

Is there a simple way of allowing the initial state to be set from the Javascript side?

**Comment by manzt** [comment]
The Python model definition is the source of truth. JavaScript must be loaded and executed to set a value on the Python side. One option would be to `allow_none` on the trait and set the value if `undefined` within the `initialize` lifecycle hook.

However, I don't recommend this approach because there will be timing issues where a property is `None` on the Python side if it is never set by JS.

**Comment by psychemedia** [comment]
Ok - thanks.

FWIW, my use case is running a simulator in the widget. The Python sid sends code to the siimulator, the simulator widget executes it, and then the js simulator exposes some state back for analysis.
